format/avro/codecs/codec.go

```diff
@@ -0,0 +1,45 @@
+package codecs
+
+import (
+	"fmt"
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+type CodecType int
+
+type Codec interface {
+	Decode(name string, d *decode.D)
+}
+
+func BuildCodec(s schema.SimplifiedSchema) (Codec, error) {
```

Comment:
A very subjective nitpick, as most decode methods seems short think i would return a function instead of interface and move all the decode function into BuildCodec (and use closures to pass config) and rename it to DecodeFnForType etc. But maybe logical types (don't know how they work) are more complicated?

Reply:
The interfaces were originally from overcomplicating the codecs where I had other required functions so callers could have more context. 

I've cleaned this up and swapped them over to 
```go
func(string, *decode.D)
```